### PR TITLE
ssl support modified with on, off switch, and cert files logic

### PR DESCRIPTION
**Pull request by DiegoSanjuan**
Also all the logic with cert files is now included inside puppet manifest so the scripts of RabbitMQ are clean.

### CODE REVIEW

File: manifests/config.pp [hunk 1]
```diff
@@ -4,6 +4,10 @@
 class rabbitmq::config(
   $port        = 5672,
   $sslport     = 5671,
+  $ssl         = 'off',
```
Comment (abonnet) [inline manifests/config.pp]: can we use boolean?

File: manifests/config.pp [hunk 2]
```diff
@@ -11,6 +15,32 @@
   $node_name   = undef,
 ) {
   
+  if $ssl != off {
```
Comment (abonnet) [inline manifests/config.pp]: Use a boolean here too, easier to read and debug

File: manifests/config.pp [hunk 3]
```diff
@@ -11,6 +15,32 @@
...
+    
+    exec { "configure_ssl_a":
```
Comment (abonnet) [inline manifests/config.pp]: This can be done using File construct with recursive option no?

File: manifests/config.pp [hunk 4]
```diff
@@ -11,6 +15,32 @@
...
+      command => "mkdir -p /etc/rabbitmq/ssl",
+      path    => "/sbin/:/usr/sbin/:/usr/bin/:/bin/",
+    }
+  
+    exec { "configure_ssl_b":
```
Comment (abonnet) [inline manifests/config.pp]: I don't like the fact we are downloading the file inside puppet. We are making the module completely dependent of ElasticBox implementation, which would make impossible to write tests for the module too. I would do that into the box scripts.

File: manifests/init.pp [hunk 5]
```diff
@@ -4,6 +4,10 @@
 class rabbitmq(
   $port        = 5672,
   $sslport     = 5671,
```
Comment (abonnet) [inline manifests/init.pp]: we use underscore in other variables

File: templates/rabbitmq.config.erb [hunk 6]
```diff
@@ -19,6 +19,10 @@
      {msg_store_index_module,rabbit_msg_store_ets_index},
      {queue_index_max_journal_entries,262144},
      {server_properties,[]},
+     <% if @ssl != 'on' %>
```
Comment (abonnet) [inline templates/rabbitmq.config.erb]: boolean too

**Comment by DiegoSanjuan** [comment]
Again key and certs downloads are being done in install script. As in the beginning.

File: manifests/config.pp (hunk 1 above)
Comment (DiegoSanjuan) [inline manifests/config.pp]: done

File: manifests/config.pp (hunk 2 above)
Comment (DiegoSanjuan) [inline manifests/config.pp]: done

File: manifests/config.pp (hunk 4 above)
Comment (DiegoSanjuan) [inline manifests/config.pp]: i taken out them again to our box.

File: manifests/init.pp (hunk 5 above)
Comment (DiegoSanjuan) [inline manifests/init.pp]: done

File: templates/rabbitmq.config.erb (hunk 6 above)
Comment (DiegoSanjuan) [inline templates/rabbitmq.config.erb]: done

File: manifests/config.pp (hunk 3 above)
Comment (DiegoSanjuan) [inline manifests/config.pp]: i taken out them again to our box.

**Comment by arnaud-elasticbox** [comment]
Ack

**Pull request merged**